### PR TITLE
Unsopported candles and tf

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py [hunk 1]
```diff
@@ -96,16 +97,12 @@ def subscribe_candle_feed(self, exchange_symbols):
                                  symbols=exchange_symbols,
                                  channels=[cryptofeed_constants.CANDLES],  # pylint: disable=E1101
                                  callbacks={cryptofeed_constants.CANDLES: candle_callback})  # pylint: disable=E1101
+            self.logger.debug(f"Subscribed to the {time_frame.value} time frame for {', '.join(exchange_symbols)}")
 
     def subscribe_feeds(self):
-        exchange_symbols = [
-            self.get_exchange_pair(pair)
-            for pair in self.pairs
-        ]
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:104]: already done here https://github.com/Drakkar-Software/OctoBot-Trading/blob/master/octobot_trading/exchanges/abstract_websocket_exchange.py#L58

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py [hunk 2]
```diff
@@ -117,15 +114,45 @@ def subscribe_feeds(self):
...
+        filtered_pairs = []
+        for pair in self.pairs:
+            if pair in exchange.normalized_symbol_mapping:
+                filtered_pairs.append(pair)
+            else:
+                formatted_pair = pair.replace(self.CRYPTOFEED_DEFAULT_MARKET_SEPARATOR, octobot_commons.MARKET_SEPARATOR)
+                self.logger.error(f"{formatted_pair} pair is not supported by this exchange's websocket")
+        self.pairs = filtered_pairs
+
+    def _ensure_time_frames(self, exchange):
```
Comment (Herklos) [inline octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:140]: I think we should rename it filter_exchange_time_frames or adapt_exchange_time_frames

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py [hunk 3]
```diff
@@ -117,15 +114,45 @@ def subscribe_feeds(self):
...
+                    self.logger.error(f"{time_frame.value} time frame is not supported by this exchange's websocket")
+            self.time_frames = filtered_time_frames
 
     def start(self):
         try:
+            self._ensure_feed_parameters()
             self.subscribe_feeds()
         except Exception as e:
-            self.logger.error(f"Failed to subscribe when creating websocket feed : {e}")
+            self.logger.exception(e, True, f"Failed to subscribe when creating websocket feed : {e}")
```
Comment (Herklos) [inline octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:155]: Why ?

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py [hunk 4]
```diff
@@ -117,15 +114,45 @@ def subscribe_feeds(self):
...
+        self._ensure_symbols(exchange)
+        self._ensure_time_frames(exchange)
+
+    def _ensure_symbols(self, exchange):
+        filtered_pairs = []
+        for pair in self.pairs:
+            if pair in exchange.normalized_symbol_mapping:
+                filtered_pairs.append(pair)
+            else:
+                formatted_pair = pair.replace(self.CRYPTOFEED_DEFAULT_MARKET_SEPARATOR, octobot_commons.MARKET_SEPARATOR)
```
[the remark this answers is not in the document]
Comment (Herklos) [reply octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:136]: I just noticed `get_pair_from_exchange` is already implemented, why not use it?

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py (hunk 3 above)
Comment (GuillaumeDSM) [reply octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:155]: To have the traceback when something goes wrong as it's manly here that something can happen and it's easier to debug later (and changes nothing for the user)

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (GuillaumeDSM) [reply octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:136]: Indeed, my bad !

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py [hunk 5]
```diff
@@ -117,15 +114,48 @@ def subscribe_feeds(self):
...
+    def _filter_exchange_time_frames(self, exchange):
+        try:
+            filtered_time_frames = []
+            for time_frame in self.time_frames:
+                if time_frame.value in exchange.valid_candle_intervals:
+                    filtered_time_frames.append(time_frame)
+                else:
+                    self.logger.error(f"{time_frame.value} time frame is not supported by this exchange's websocket")
+            self.time_frames = filtered_time_frames
+        except AttributeError:
```
Comment (Herklos) [inline octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:149]: :+1:

**Review by Herklos — APPROVED** [review]

**Pull request merged**